src/iam/client/iamclient.hpp

```diff
@@ -0,0 +1,151 @@
+     * @returns aos::Error.
+     */
+    aos::Error Init(const Config& config, aos::iam::certhandler::CertHandlerItf& certHandler,
+        aos::cryptoutils::CertLoaderItf& certLoader, aos::crypto::x509::ProviderItf& cryptoProvider,
+        bool provisioningMode);
+
+    /**
+     * Returns remote node identifiers.
+     *
+     * @result aos::Array<aos::StaticString<aos::cNodeIDLen>>
+     */
+    aos::Array<aos::StaticString<aos::cNodeIDLen>> GetRemoteNodes() override;
+
+    /**
+     * Returns IAM cert types.
+     *
+     * @param nodeID node id.
+     * @param[out] certTypes result certificate types.
+     * @returns aos::Error.
+     */
+    aos::Error GetCertTypes(const aos::String&                              nodeID,
+        aos::Array<aos::StaticString<aos::iam::certhandler::cCertTypeLen>>& certTypes) override;
+
+    /**
+     * Owns security storage.
+     *
+     * @param nodeID node id.
+     * @param certType certificate type.
+     * @param password owner password.
+     * @returns aos::Error.
+     */
+    aos::Error SetOwner(const aos::String& nodeID, const aos::String& certType, const aos::String& password) override;
+
+    /**
+     * Clears security storage.
+     *
+     * @param nodeID node id.
+     * @param certType certificate type.
+     * @returns aos::Error.
+     */
+    aos::Error Clear(const aos::String& nodeID, const aos::String& certType) override;
+
+    /**
+     * Creates key pair.
+     *
+     * @param nodeID node id.
+     * @param certType certificate type.
+     * @param subjectCommonName common name of the subject.
+     * @param password owner password.
+     * @param[out] pemCSR certificate signing request in PEM.
+     * @returns aos::Error.
+     */
+    aos::Error CreateKey(const aos::String& nodeID, const aos::String& certType, const aos::String& subjectCommonName,
+        const aos::String& password, aos::String& pemCSR) override;
+
+    /**
+     * Applies certificate.
+     *
+     * @param nodeID node id.
+     * @param certType certificate type.
+     * @param pemCert certificate in a pem format.
+     * @param[out] info result certificate information.
+     * @returns aos::Error.
+     */
+    aos::Error ApplyCertificate(const aos::String& nodeID, const aos::String& certType, const aos::String& pemCert,
+        aos::iam::certhandler::CertInfo& info) override;
+
+    /**
+     * Encrypts disk for a node.
+     *
+     * @param nodeID node identifier.
+     * @param password password.
+     * @returns aos::Error.
+     */
+    aos::Error EncryptDisk(const aos::String& nodeID, const aos::String& password) override;
+
+    /**
+     * Finishes provisioning.
+     *
+     * @param nodeID node identifier.
+     * @returns aos::Error.
+     */
+    aos::Error FinishProvisioning(const aos::String& nodeID) override;
+
+protected:
+    virtual CertificateServiceStubPtr  CreateIAMCertificateServiceStub(const aos::String& nodeId);
```

Comment:
Why do we need it protected and virtual?

Reply:
To mock grpc connection in UT(s)